### ISSUE TITLE
Let zip_fseek() clear the EOF flag when appropriate

### ISSUE THREAD

**Issue by Stockentenbrot** [nih-at/libzip#310]
**Description**
When reading past EOF using `zip_fread()` subsequent calls read nothing, just as defined in the documentation. A following `zip_fseek() `to the beginning of the file does not seem to clear the EOF flag and subsequent calls to `zip_fread()` still return with zero bytes read.

This turned out to be a problem for me when I needed to probe the file types of tiny (around 1 kbyte) uncompressed files (store method) in a .zip archive. I have to probe multiple possible formats by reading headers (the file extension is of no use in this case), but in some cases for certain possible formats a probing read can be larger than the file size, resulting in a read that hits the EOF. Further probes are then not possible since from then on `zip_fread()` always reads zero bytes.

**Solution**
I'd propose to make `zip_fseek()` clear the internal EOF flag where appropriate or to provide a new function similar to POSIX `rewind()`

**My workaround for now**
I worked around this by checking the file size against the requested read bytes before each probe to make sure that I will never hit the EOF, but I feel that the proposed behavior would be more user friendly, especially since this is the way the POSIX equivalent `fseek()` behaves.

**Comment by dillof** [comment]
We can't reproduce this behaviour in the current version of libzip. We added a test that demonstrates this: `read_seek_read.test`.

We had this in prior versions, but fixed it in 2018. If you are using a newer version, please post an example program that demonstrates your described failure.

**Comment by Stockentenbrot** [comment]
I just tried it with v1.9.2 and it is indeed fixed. I was using the distro supplied version 1.5.1 before where it didn't work. In both cases the first read returns the entire file while the following call returns 0 bytes.
See the attached test:
[libzipfseektest.zip](https://github.com/nih-at/libzip/files/9190931/libzipfseektest.zip)
I've also checked with v1.7.3 from the current Debian stable where even the first call doesn't return anything.
Do you know the version number after which that bug was fixed and behavior like in v1.9.2 can be expected so that I can implement a check for it in my code?

**Comment by 0-wiz-0** [comment]
The EOF bug you mention was fixed in 1.5.2.
I'm not sure what problem you see in 1.7.3, I can not reproduce it:
```
./fseektest                                                                
libzip version: 1.7.3
read: 27
read: 8
```
